Assets/MixedRealityToolkit.Providers/ObjectMeshObserver/SpatialObjectMeshObserver.cs

```diff
@@ -0,0 +1,336 @@
+
+                    ApplyMeshMaterial(meshObject);
+
+                    meshes.Add(currentMeshId, meshObject);
+
+                    currentMeshId++;
+                }
+            }
+            sendObservations = false;
+        }
+
+        /// <inheritdoc />
+        public override void Suspend()
+        {
+            IsRunning = false;
+        }
+
+        private void RemoveMeshObject(int id)
+        {
+            SpatialAwarenessMeshObject meshObject = null;
+
+            if (meshes.TryGetValue(id, out meshObject))
+            {
+                // Remove the mesh object from the collection.
+                meshes.Remove(id);
+                if (meshObject != null)
+                {
+                    SpatialAwarenessMeshObject.Cleanup(meshObject);
+                }
+
+                // Send the mesh removed event
+                SpatialAwarenessSystem?.RaiseMeshRemoved(this, id);
+            }
+        }
+
+        #endregion IMixedRealitySpatialAwarenessObserver implementation
+
+        #region IMixedRealitySpatialAwarenessMeshObserver implementation
+
+        public SpatialAwarenessMeshDisplayOptions displayOption = SpatialAwarenessMeshDisplayOptions.Visible;
+        
+        /// <inheritdoc />
+        public SpatialAwarenessMeshDisplayOptions DisplayOption
+        {
+            get => displayOption;
+
+            set
+            {
+                displayOption = value;
+                ApplyUpdatedMeshDisplayOption(displayOption);
+            }
+        }
+
+        /// <inheritdoc />
+        public SpatialAwarenessMeshLevelOfDetail LevelOfDetail { get; set; } = SpatialAwarenessMeshLevelOfDetail.Coarse;
+
+        private Dictionary<int, SpatialAwarenessMeshObject> meshes = new Dictionary<int, SpatialAwarenessMeshObject>();
+
+        /// <inheritdoc />
+        public IReadOnlyDictionary<int, SpatialAwarenessMeshObject> Meshes =>
+            new Dictionary<int, SpatialAwarenessMeshObject>(meshes) as IReadOnlyDictionary<int, SpatialAwarenessMeshObject>;
+
+        private int meshPhysicsLayer = 31;
+
+        /// <inheritdoc />
+        public int MeshPhysicsLayer
+        {
+            get => meshPhysicsLayer;
+
+            set
+            {
+                if ((value < 0) || (value > 31))
+                {
+                    Debug.LogError("Specified MeshPhysicsLayer is out of bounds. Please use a value between 0 and 31, inclusive.");
+                    return;
+                }
+                meshPhysicsLayer = value;
+            }
+        }
+
+        /// <inheritdoc />
+        public int MeshPhysicsLayerMask => (1 << MeshPhysicsLayer);
+
+        /// <inheritdoc />
+        public bool RecalculateNormals { get; set; } = true;
+
+        /// <inheritdoc />
+        public int TrianglesPerCubicMeter { get; set; } = 0;
+
+        /// <inheritdoc />
+        public Material OcclusionMaterial { get; set; } = null;
```

Comment:
If the mesh is already displaying the occlusion material and this property is changed, how will the mesh get updated to the new material?

Reply:
you update the display option. we _can_ change the semantic.

Reply:
ex:

observer.OcclusionMaterial = newMaterial;
observer.DisplayOption = Occlusion;